mesonbuild/build.py

```diff
@@ -1090,14 +1101,22 @@ def get_arglist(self, inname):
         basename = os.path.splitext(plainname)[0]
         return [x.replace('@BASENAME@', basename).replace('@PLAINNAME@', plainname) for x in self.arglist]
 
+    def is_parent_path(self, parent, trial):
+        relpath = os.path.relpath(os.path.normpath(trial), os.path.normpath(parent))
+        return not relpath.startswith('..') # For subdirs we can only go "down".
```

Comment:
I would recommend looking at `pathlib.PurePath` and using that for all path manipulation. It's much simpler. F.ex, this would just be:
```py
relpath = PurePath(trial).relative_to(parent)
return relpath.parts[0] != '..'
```
`PurePath` normalizes automatically.